hibernate-core/src/main/java/org/hibernate/type/descriptor/java/UUIDJavaType.java

```diff
@@ -41,8 +41,8 @@ public String toString(UUID value) {
 		return ToStringTransformer.INSTANCE.transform( value );
 	}
 
-	public UUID fromString(CharSequence string) {
-		return ToStringTransformer.INSTANCE.parse( string.toString() );
+	public UUID fromString(CharSequence string)  {
+		return '-' == string.charAt( 8 ) ? ToStringTransformer.INSTANCE.parse( string.toString() ) : NoDashesStringTransformer.INSTANCE.parse( string.toString() );
```

Comment:
I don't mind this change, but is it necessary to get some test working?

Reply:
Yes, the new UUID type is causing some issues with json in oracle; this changes fixes a number of tests in `JsonEmbeddableTest`, but there is still one remaining (`testSelectionItems`), that as far as i can tell has to with the function hexToRaw not liking that the UUID contain '-'.